lms/djangoapps/completion/tests/test_models.py

```diff
@@ -0,0 +1,100 @@
+"""
+Test models, managers, and validators.
+"""
+
+from django.core.exceptions import ValidationError
+from django.test import TestCase
+from opaque_keys.edx.keys import UsageKey
+
+from student.tests.factories import UserFactory
+
+from .. import models
+
+
+class ValidatorTestCase(TestCase):
+    """
+    Test that validate_ratio only allows floats (and ints) between 0.0 and 1.0.
+    """
+    def test_valid_ratios(self):
+        for value in [1.0, 0.0, 1, 0, 0.5, 0.333081348071397813987230871]:
+            models.validate_ratio(value)
+
+    def test_invalid_ratios(self):
+        for value in [-0.00000000001, 1.0000000001, 47.1, 1000, 'hi', None]:
+            self.assertRaises(ValidationError, models.validate_ratio, value)
```

Comment:
It currently allows `True` and `False` through without error - does that matter?

Reply:
I don't think so.  This is only going to be applied to FloatFields.  None is a possible value in a float field, but if you pass store true or false in a FloatField, they will just get stored as 1.0 and 0.0, (because bool is a subclass of float).  In fact, there's really no reason to test 'hi'.  A (slightly) more interesting question would be how the validator handles float('inf') and float('nan').